unidist/config/parameter.py

```diff
@@ -174,7 +174,12 @@ def put(cls, value):
         """
         if not _TYPE_PARAMS[cls.type].verify(value):
             raise ValueError(f"Unsupported value: {value}")
-        cls._value = _TYPE_PARAMS[cls.type].normalize(value)
+
+        value = _TYPE_PARAMS[cls.type].normalize(value)
+        if cls.choices is not None and value in cls.choices:
```

Comment:
Good addition but let's introduce the changes as part of separate PR. And it would probably better to add a check in .get() because a user can set value via an env var too.

Reply:
Ok. It will be done in #49.